packages/react-components/src/components/utils/responsive.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { RefObject, useEffect, useState } from 'react';
+
+/**
+ * A utility hook for providing the width of a parent element.
+ * Returns updated width if parent/window resizes.
+ * @param containerRef - Ref of a parent element whose width will be returned.
+ */
+export const useContainerWidth = (containerRef: RefObject<HTMLDivElement>): number => {
+  const [width, setWidth] = useState(0);
+  useEffect(() => {
+    const updateWidth = (): void => {
+      const width = containerRef.current?.offsetWidth ?? 0;
+      setWidth(width);
+    };
+    updateWidth();
+    window.addEventListener('resize', updateWidth);
+    return () => {
+      window.removeEventListener('resize', updateWidth);
+    };
+  }, [containerRef]);
+
+  return width;
+};
+
+/**
+ * A utility hook for identifying if the current viewport size is is mobile screen.
+ * Returns updated result if parent/window resizes.
+ * @param containerRef - Ref of a parent element whose width will be used to determine the viewport size
+ */
+export const useIsMobileScreen = (containerRef: RefObject<HTMLDivElement>): boolean => {
```

Comment:
can we abstract terms of `mobile` away and just have `isNarrowContainer` - the width of a screen doesn't actually determine if is a mobile screen.
Maybe even just have a helper function instead of a whole react hook
```
isNarrow = (containerSize) => containerWidth <= NARROW_WIDTH
```

Reply:
`Narrow` becomes too generic a term here. 
I am thinking `smallScreen` should be a better choice here. Thoughts?
For tablets we can use `mediumScreen`

Well that's how bootstrap does it.
`xs` for mobile, `sm` for handheld devices, `md` for low res desktop, etc...

Reply:
I like smallScreen etc. (though technically its not `screen` but whatever container we're in.. but screen is probably sufficient - maybe smallContainer? But I don't really like that)

Reply:
So for our component, it doesn't know or care about whether it's in a screen or container..
We can stick to a uniform naming assuming that we are talking about size w.r.t screen size.

Reply:
This was marked resolved but it is still isMobile everywhere - unresolving